samza-core/src/main/java/org/apache/samza/runtime/AbstractApplicationRunner.java

```diff
@@ -106,4 +112,25 @@ final ExecutionPlan getExecutionPlan(StreamApplication app) throws Exception {
   final StreamManager getStreamManager() {
     return streamManager;
   }
+
+  /**
+   * Write the execution plan JSON to a file
+   * @param planJson JSON representation of the plan
+   */
+  final void writePlanJsonFile(String planJson) {
+    try {
+      String content = "plan='" + planJson + "'";
+      String binPath = System.getenv(ShellCommandConfig.JOB_BIN_DIR());
```

Comment:
I just feel that it's unconventional to write to anything to bin. I wouldn't even look there for output of an application.

If it must be a static location, I suggest creating a dir called "execution_plan" and put the files there.